paper-server/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java

```diff
@@ -2122,6 +2128,69 @@ public static boolean callEntityInteractEvent(Entity nmsEntity, Block block) {
         return !event.isCancelled();
     }
 
+    public static List<BlockPos> handleExplodeEvent(ServerExplosion serverExplosion, List<BlockPos> blockPositions) {
+        // First convert the blockPositionList related to explosion to Bukkit
+        World bworld = serverExplosion.level().getWorld();
+
+        List<Block> blockList = new ObjectArrayList<>();
+        for (int posBlockPositionList = blockPositions.size() - 1; posBlockPositionList >= 0; posBlockPositionList--) {
+            org.bukkit.block.Block bblock = CraftBlock.at(serverExplosion.level(), blockPositions.get(posBlockPositionList));
+            if (!bblock.isEmpty() || serverExplosion.fire) {
```

Comment:
I still don't like this workaround, you don't have to fix the issue in this PR, but having an inconsistent block list is even worse for plugins. Can you just revert to the empty check for now?

Reply:
Hmmm but this make all the events cause that bug... I feel better keep this.

Reply:
reverted since you are unable to fix it properly and i don't want to spoonfeed